IanGRap/Squish
# Squish detection bugfix

Goal:
Fix the bug where the player would sometimes die when moving quickly, even if they weren't squished


Important Files Changed:
- The Player prefab:
    - Instead of 1 squish collider, there are 4. (Top, Bottom, Right and Left)
- SquishDetector.cs


What Changed:
- Squish detection needed to be a lot more complex to fix this bug. I added in the following checks before death:
    - (Top + Bottom) colliders or (Right + Left) colliders must be activated in tandem in order to trigger a squish
    - The player must be colliding with at least 2 _unique_ objects in order to trigger a squish
    - The player will not be squished if they are between 2 objects that are less than the Min Squish Distance apart (1.5 units at this time)
        - This prevents the scenario of the player dying when Unity physics clips them directly between 2 blocks that are clumped together
    - The player cannot be in the middle of a dash in order to trigger a squish


How to Test:
- Open Assets/Scenes/demos/sprint 1
- Dash into blocks from all angles, trying to cause a player death without actually being squished between 2 objects
- Make sure that you do still get squished when you _are_ supposed to!


Note: 
- I played the scene all the way through about 15 times. Out of those tests, I got only 1 questionable death. This squish detection is definitely more accurate but still (I think) not perfect. However, I'm fresh out of ideas of how to make it more perfect. Perhaps someone else will have some ideas. I'm definitely open to suggestions.
